go/common/errutil/errors_util.go

```diff
@@ -12,3 +12,29 @@ var (
 	ErrNotFound = ethereum.NotFound
 	ErrNoImpl   = errors.New("not implemented")
 )
+
+// InternalError is the implementation of the SystemError interface that's used for system consumption only
+type InternalError struct {
+	msg string
+	err error
+}
+
+func NewInternalErr(err error) error {
```

Comment:
if we rename the package syserr, then the constructor could be "New", making things more readable everywhere